src/util/replace_symbol.cpp

```diff
@@ -51,7 +51,10 @@ bool replace_symbolt::replace_symbol_expr(symbol_exprt &s) const
     s.type() == it->second.type(),
     "types to be replaced should match. s.type:\n" + s.type().pretty() +
       "\nit->second.type:\n" + it->second.type().pretty());
+  source_locationt previous_source_location{s.source_location()};
   static_cast<exprt &>(s) = it->second;
+  if(previous_source_location.is_not_nil())
```

Comment:
If this  location transplant really must happen, maybe  it can be achieved by manually copying the location to the replacement expression before building the replacement map ?

Reply:
just looking at it again, previous_source_location comes from the symbol_exprt that's being replaced